service/service_setup.go

```diff
@@ -34,21 +34,46 @@ import (
 	"go.uber.org/zap"
 )
 
-func (svc *serviceCore) setupLogging() {
-	err := svc.configProvider.Get("logging").PopulateStruct(&svc.logConfig)
-	if err != nil {
-		zap.L().Info("Logging configuration is not provided, using UberFX defaults", zap.Error(err))
+func (svc *serviceCore) setupLogging() error {
+	cfg := svc.configProvider.Get("logging")
+	if cfg.HasValue() {
+		// Uhhh... this process is not the most elegant.
+		// Becuase log.Configuration embeds zap, the PopulateStruct
+		// does not work properly as it's unable to serialize fields directly
+		// into the embedded struct, so inner struct has to be treated as a
```

Comment:
I think those tests probably cover naming the structs explicitly, ie

given

```
type struct logConfig{
   zap.Config
   Sentry
}
```

The following would work:

```
logging:
   config:
      foo: bar
   sentry:
      dsn: asdf
```

But not this (flat):

```
logging:
   foo: bar
   sentry:
      dsn: asdf
```